src/details/ArborX_Box.hpp

```diff
@@ -25,79 +25,85 @@ namespace ArborX
  * size 2x spatial dimension with a default constructor to initialize
  * properly an "empty" box.
  */
-struct Box
+template <int DIM = 3>
+struct BoxD
```

Comment:
Should probably put all `*D` into Experimental namespace or something.

Reply:
I would try.  We might get surprises with ADL